web/src/debug.ts

```diff
@@ -10,20 +10,26 @@
     evaluates to foo() and prints the elapsed time
     to the console along with the name "foo". */
 
-export function print_elapsed_time(name, fun) {
+type Collision = {
+    id: string;
+    count: number;
+    node: string;
+};
+
+export function print_elapsed_time<T>(name: string, fun: () => T): T {
     const t0 = Date.now();
     const out = fun();
     const t1 = Date.now();
-    console.log(name + ": " + (t1 - t0) + " ms");
+    console.log(`${name} : ${t1 - t0} ms`);
     return out;
 }
 
-export function check_duplicate_ids() {
-    const ids = new Set();
-    const collisions = [];
+export function check_duplicate_ids(): {collisions: Collision[]; total_collisions: number} {
+    const ids = new Set<string>();
+    const collisions: Collision[] = [];
     let total_collisions = 0;
 
-    for (const o of Array.prototype.slice.call(document.querySelectorAll("*"))) {
+    for (const o of document.querySelectorAll("*")) {
```

Comment:
Why does this remove `Array.prototype.slice.call`?  This seems like it could be a functional change.

Reply:
Oh, I see this is discussed above. This information belongs in the commit message, not just in the discussion thread.